tests/test_random.py

```diff
@@ -0,0 +1,4 @@
+from src.random_number import generate
+
+def test_random() -> None:
+    assert generate() <= 10 and generate() >= 1
```

Comment:
Così stai chiamando due volte la funzione `generate`, per cui stai testando due valori diversi

Reply:
Quindi era meglio usare il mocker?

Reply:
nel tuo caso puoi mettere in una variabile il valore ritornato dal generate() e testare se il valore di quella variabile è appartenente al range. Se lasci il test in quel modo, e riscrivendo il generate() in modo sbagliato, come ad esempio
```
def generate() -> int:
    return random.randint(1,1000)
```
il tuo test potrebbe passare perchè quella funzione può generare nella prima condizione un 5 (che soddisfa <= 10) e un 50 (che soddisfa >=1)

Reply:
ma anche fare
```python
assert  10  <= generate() <= 1
```
non è male

Idealmente sarebbe meglio usare la variabile mocker per spiare la funzione randint() e verificare che essa venga chiamata e viene chiamata con i valori 1,1000 usando `spy.called_with(1,1000)`